logtail/logtail.go

```diff
@@ -659,10 +660,12 @@ func (l *Logger) Write(buf []byte) (int, error) {
 			l.stderr.Write(withNL)
 		}
 	}
+
 	l.writeLock.Lock()
-	b := l.encode(buf, level)
-	_, err := l.send(b)
-	l.writeLock.Unlock()
+	defer l.writeLock.Unlock()
+
+	b := l.encodeLocked(buf, level)
+	_, err := l.sendLocked(b)
```

Comment:
should this really be done while holding the lock? (not suggesting you change it in this pr, just asking)

Reply:
It doesn't really do much, but I didn't put much thought into it yet.

The filch implementation of Buffer and the memory implementation have different ownership lifetimes. Need to figure out a design that works for both of them, or unify and remove the interface. Or something.